Neos.ContentRepository.Core/Classes/Service/ContentRepositoryMaintainer.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\Service;
+
+use Neos\ContentRepository\Core\Factory\ContentRepositoryServiceInterface;
+use Neos\ContentRepository\Core\Feature\ContentStreamEventStreamName;
+use Neos\ContentRepository\Core\Feature\WorkspaceEventStreamName;
+use Neos\ContentRepository\Core\Subscription\Engine\Errors;
+use Neos\ContentRepository\Core\Subscription\Engine\SubscriptionEngine;
+use Neos\ContentRepository\Core\Subscription\Engine\SubscriptionEngineCriteria;
+use Neos\ContentRepository\Core\Subscription\SubscriptionStatuses;
+use Neos\ContentRepository\Core\Subscription\SubscriptionId;
+use Neos\Error\Messages\Error;
+use Neos\EventStore\EventStoreInterface;
+use Neos\EventStore\Model\Event\EventType;
+use Neos\EventStore\Model\Event\EventTypes;
+use Neos\EventStore\Model\Event\StreamName;
+use Neos\EventStore\Model\EventStore\Status as EventStoreStatus;
+use Neos\EventStore\Model\EventStream\EventStreamFilter;
+use Neos\EventStore\Model\EventStream\VirtualStreamName;
+
+/**
+ * API to set up and manage a content repository
+ *
+ * @api
+ */
+final readonly class ContentRepositoryMaintainer implements ContentRepositoryServiceInterface
+{
+    /**
+     * @internal please use the {@see ContentRepositoryMaintainerFactory} instead!
+     */
+    public function __construct(
+        private EventStoreInterface $eventStore,
+        private SubscriptionEngine $subscriptionEngine
+    ) {
+    }
+
+    public function setUp(): Error|null
+    {
+        $this->eventStore->setup();
+        $eventStoreIsEmpty = iterator_count($this->eventStore->load(VirtualStreamName::all())->limit(1)) === 0;
+        $setupResult = $this->subscriptionEngine->setup();
+        if ($setupResult->errors !== null) {
+            return self::createErrorForReason('setup', $setupResult->errors);
+        }
+        if ($eventStoreIsEmpty) {
+            // todo reintroduce skipBooting flag, and also notify if the flag is not set, e.g. because there are events
+            $bootResult = $this->subscriptionEngine->boot();
```

Comment:
with `skipBooting` i mean this flag https://github.com/patchlevel/event-sourcing/blob/dae05647290ae5e76a92d7ba9a1ca0284cc5003c/src/Subscription/Engine/DefaultSubscriptionEngine.php#L48

but maybe the `boot` case is more explicit and its only minor overhead, and less complex api surface for us.